pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -349,9 +349,12 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 							To: routev1.RouteTargetReference{
 								Kind: "Service",
 								Name: infinispan.Name},
-							TLS: &routev1.TLSConfig{Termination: routev1.TLSTerminationPassthrough},
 						},
 					}
+					if infinispan.Spec.Security.EndpointEncryption.Type != "None" &&
```

Comment:
@rigazilla you can use CertificateSourceType with latest backport, otherwise CR validation could be failed